File: EPRO.Library.v3.5/EPRO.Library.v3.5/Objects/EInt.cs
using System;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace ELibrary.Standard.Objects
{
    public sealed class EInt
    {

        /// <summary>
        /// Converts An Object to long
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int valueOf(string obj)
        {
            try
            {
                if (Information.IsNothing(obj))
                    return 0;
                if (string.IsNullOrEmpty(obj))
                    return 0;
                if (string.IsNullOrEmpty(obj.Trim()))
                    return 0;
                return Conversions.ToInteger(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }


        /// <summary>
        /// Converts An Object to int32
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int valueOf(DBNull obj)
        {
            return 0;
        }





        /// <summary>
        /// Converts An Object to int32
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int valueOf(double obj)
        {
            try
            {
                return (int)Math.Round(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }

        /// <summary>
        /// Converts An Object to int32
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int valueOf(decimal obj)
        {
            try
            {
                return (int)Math.Round(obj);
            }
            catch (Exception ex)
            {
                return 0;
            }
        }




        /// <summary>
        /// Converts objects to integer
        /// </summary>
        /// <param name="obj"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int valueOf(object obj)
        {
            if (obj is null)
                return 0;
            if (obj is string)
                return valueOf(Conversions.ToString(obj));
            if (obj is DBNull)
                return valueOf((DBNull)obj);
            if (obj is double)
                return valueOf(Conversions.ToDouble(obj));
            if (obj is long)
                return valueOf((decimal)Conversions.ToLong(obj));
            if (obj is int)
                return Conversions.ToInteger(obj);
            if (obj is short)
                return Conversions.ToInteger(obj);
            if (obj is decimal)
                return valueOf(Conversions.ToDecimal(obj));
            if (obj is bool)
                return EShort.valueOf(Conversions.ToBoolean(obj));
            return 0;    // REM Cant convert this
        }
    }
}

File: EPRO.Library.v3.5/EPRO.Library.v3.5/Objects/EInt.vb
Option Strict On
Option Explicit On

Namespace Objects

    Public NotInheritable Class EInt

        ''' <summary>
        ''' Converts An Object to long
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As String) As Int32
            Try

                If IsNothing(obj) Then Return 0
                If obj = "" Then Return 0
                If obj.Trim = "" Then Return 0

                Return CInt(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function


        ''' <summary>
        ''' Converts An Object to int32
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As System.DBNull) As Int32

            Return 0

        

        End Function





        ''' <summary>
        ''' Converts An Object to int32
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Double) As Int32
            Try

                Return CInt(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function

        ''' <summary>
        ''' Converts An Object to int32
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Decimal) As Int32
            Try

                Return CInt(obj)

            Catch ex As Exception
                Return 0

            End Try

        End Function




        ''' <summary>
        ''' Converts objects to integer
        ''' </summary>
        ''' <param name="obj"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        Public Overloads Shared Function valueOf(ByVal obj As Object) As Int32

            If obj Is Nothing Then Return 0

            If TypeOf obj Is String Then Return valueOf(CType(obj, String))
            If TypeOf obj Is DBNull Then Return valueOf(CType(obj, DBNull))
            If TypeOf obj Is Double Then Return valueOf(CType(obj, Double))
            If TypeOf obj Is Long Then Return valueOf(CType(obj, Long))
            If TypeOf obj Is Int32 Then Return CInt(obj)
            If TypeOf obj Is Int16 Then Return CInt(obj)
            If TypeOf obj Is Decimal Then Return valueOf(CType(obj, Decimal))
            If TypeOf obj Is Boolean Then Return EShort.valueOf(CType(obj, Boolean))

            Return 0    REM Cant convert this

        End Function



    End Class

End Namespace